content/sklearn/concepts/nearest-neighbors/nearest-neighbors.md

```diff
@@ -0,0 +1,115 @@
+        43.1, 45.3, 45.7, 47.5, 47.2, 46.9, 45, 42.3, 45.1, 46.1,
+        47.1, 45.6, 45.7, 45.1, 44.9, 43.5, 46.1, 45.3, 45, 45],
+    "Season": [
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+}
+
+# Convert to DataFrame
+df = pd.DataFrame(data)
+
+# Preprocess data
+X = df[["Snow Depth (in)", "Air Temperature Average (degF)", "Snow Density (pct)"]]
+y = df["Season"]
+
+# Classification model
+knn_classifier = KNeighborsClassifier(n_neighbors=5)
+knn_classifier.fit(X, y)
+
+X_test = np.array([[29, 27, 45.5]])
+
+predictions = knn_classifier.predict(X_test)
+
+print(f'Predicted season for the given snow condition is: {["Accumulation" if predictions[0]==1 else "Melt"][0]}.')
+```
+
+### Usage of Nearest Neighbors in Scikit-learn for a **regression** problem:
+The following code predicts the discharge in cubic feet per second based on the snow water equivalent in inches using a k-Nearest Neighbors (k-NN) regression model. The snowmax is an array representing snow water equivalent measurements, and flowmax, is an array representing corresponding discharge values. The model is trained using these arrays. The parameter `n_neighbors=5` specifies that the prediction for a new data point will be based on the **average discharge values** of the 5 nearest neighbors in the training data. The search algorithm implicitly used by `KNeighborsRegressor` finds the nearest neighbors by calculating the distance between points in the feature space, typically using the Euclidean distance. The code then predicts the discharge for a new snow water equivalent value of 42 inches and prints the result. This method outputs a value that is the averages the output, leading to potentially new values not seen in the training set.
+
+```shell
+import numpy as np
+from sklearn.neighbors import KNeighborsRegressor
+
+# Snow water equivalent in inch
+snowmax = np.array([44.3, 43.6, 21.1, 58.4, 48.5, 51. , 38.7, 53.3, 16.8, 27. , 46.4,
+       24.6, 29.7, 16.9, 41.2, 28.7, 34.9, 42.7, 55.9, 45. , 49.6, 34.7,
+       25.4, 29.8, 34.5, 31.5, 45.2, 58.2, 26.1, 42. , 42.1, 28.9, 66.2,
+       29.7, 24.2, 48.5, 27.6, 37.6, 54.2, 32.3, 41.5, 41.7, 26.4])  
+
+# Discharge in cubic feet per second
+flowmax = np.array([ 855.,  876., 584., 1270., 1400., 1830., 865., 1870.,  338.,
+        540.,  748., 290.,  760., 243., 1180., 595.,  1220., 1330.,
+        1560., 977., 1430., 562., 721., 492.,  716.,  529., 1230.,
+        1530., 468., 722.,  967., 852., 1630., 429.,  480., 773.,
+        545.,  775., 1390., 628., 929., 695.,  271.])
+
+snow_test = np.array([42])
+
+# Regression model
+knn_regressor = KNeighborsRegressor(n_neighbors=5)
+knn_regressor.fit(snowmax.reshape(-1, 1), flowmax)
+predictions = knn_regressor.predict(snow_test.reshape(-1, 1))
+
+print(f'Predicted flow for snow water equivalent of {snow_test[0]} inches: {predictions[0]} cubic feet per second.')
+```
```

Comment:
``` suggestion
```py
import numpy as np
from sklearn.neighbors import KNeighborsRegressor

# Snow water equivalent in inch
snowmax = np.array([44.3, 43.6, 21.1, 58.4, 48.5, 51. , 38.7, 53.3, 16.8, 27. , 46.4,
       24.6, 29.7, 16.9, 41.2, 28.7, 34.9, 42.7, 55.9, 45. , 49.6, 34.7,
       25.4, 29.8, 34.5, 31.5, 45.2, 58.2, 26.1, 42. , 42.1, 28.9, 66.2,
       29.7, 24.2, 48.5, 27.6, 37.6, 54.2, 32.3, 41.5, 41.7, 26.4])  

# Discharge in cubic feet per second
flowmax = np.array([ 855.,  876., 584., 1270., 1400., 1830., 865., 1870.,  338.,
        540.,  748., 290.,  760., 243., 1180., 595.,  1220., 1330.,
        1560., 977., 1430., 562., 721., 492.,  716.,  529., 1230.,
        1530., 468., 722.,  967., 852., 1630., 429.,  480., 773.,
        545.,  775., 1390., 628., 929., 695.,  271.])

snow_test = np.array([42])

# Regression model
knn_regressor = KNeighborsRegressor(n_neighbors=5)
knn_regressor.fit(snowmax.reshape(-1, 1), flowmax)
predictions = knn_regressor.predict(snow_test.reshape(-1, 1))

print(f'Predicted flow for snow water equivalent of {snow_test[0]} inches: {predictions[0]} cubic feet per second.')
```
```
The code should always be written inside their supportive code blocks but not inside the shell block. For more information check this [documentation](documentation/style-guide.md)

Reply:
Done!